control/server.py

```diff
@@ -204,6 +207,37 @@ def _start_spdk(self):
         for trtype in spdk_transports.split():
             self._create_transport(trtype.lower())
 
+    def _stop_spdk(self):
+        """Stops SPDK process."""
+        assert self.spdk_process is not None # should be verified by the caller
+
+        return_code = self.spdk_process.returncode
+        rpc_socket = self.config.get("spdk", "rpc_socket")
```

Comment:
One thing that I'm worried about using `configparser` is that most parsing errors will occur lazily, at the time the value is accessed. I know that `rpc_socket` is already fetched during start-up phase, but I'd personally enforce required params & defaults at initialization. I just [opened this issue](https://github.com/ceph/ceph-nvmeof/issues/191) to tackle that.

Reply:
Guess we're safe here. In general, digital sheep dream of Rust style safeness. I recommend beginning by establishing the schema for the configuration class, along with mandatory default values. Agree on validation of the configuration file during creation or initial conf file read to ensure a 'fail-fast' approach. While the INI format and the ('section', 'key') concept are commonly used, considering a more versatile and generic class might provide greater flexibility.